File: src/Java/databaseConnection.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

// Class to connect to the database useful for API calls
public class databaseConnection {
    private final String url;
    private final Properties properties;

    public databaseConnection(String host, String databaseName, String username, String password) {

        this.url = "jdbc:postgresql://" + host + "/" + databaseName;
        this.properties = new Properties();
        this.properties.setProperty("user", username);
        this.properties.setProperty("password", password);
    }

    public Connection getConnection() throws SQLException {
        return DriverManager.getConnection(this.url, this.properties);
    }
}

File: src/Python/databaseConnection.py
# class to connect to the database and execute queries useful for API calls

class DatabaseConnection:
    def __init__(self):
        self.connection = None
        self.cursor = None

    
    # Connect to the database
    def connect(self):
        try:
            self.connection = psycopg2.connect(
                dbname="eventmanager",
                user="postgres",
                host="localhost",
                password="password"
            )
            self.cursor = self.connection.cursor()
        except Exception as e:
            print("Error connecting to the database: ", e)

    # Close the connection
    def close(self):
        self.cursor.close()
        self.connection.close()

    # Execute a query
    def execute(self, query):
        try:
            self.cursor.execute(query)
            self.connection.commit()
        except Exception as e:
            print("Error executing query: ", e)

    # Fetch the results of a query
    def fetch(self):
        return self.cursor.fetchall()
